packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/DuplicateRepresentationEventHandler.java

```diff
@@ -0,0 +1,161 @@
+import org.eclipse.sirius.components.collaborative.api.ChangeKind;
+import org.eclipse.sirius.components.collaborative.api.IEditingContextEventHandler;
+import org.eclipse.sirius.components.collaborative.api.IRepresentationMetadataPersistenceService;
+import org.eclipse.sirius.components.collaborative.api.IRepresentationPersistenceService;
+import org.eclipse.sirius.components.collaborative.api.IRepresentationSearchService;
+import org.eclipse.sirius.components.collaborative.api.Monitoring;
+import org.eclipse.sirius.components.collaborative.messages.ICollaborativeMessageService;
+import org.eclipse.sirius.components.core.api.ErrorPayload;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IInput;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.components.representations.Failure;
+import org.eclipse.sirius.components.representations.IRepresentation;
+import org.eclipse.sirius.components.representations.IStatus;
+import org.eclipse.sirius.components.representations.Message;
+import org.eclipse.sirius.components.representations.MessageLevel;
+import org.eclipse.sirius.components.representations.Success;
+import org.eclipse.sirius.web.application.UUIDParser;
+import org.eclipse.sirius.web.application.views.explorer.dto.DuplicateObjectInput;
+import org.eclipse.sirius.web.application.views.explorer.dto.DuplicateRepresentationInput;
+import org.eclipse.sirius.web.application.views.explorer.dto.DuplicateRepresentationSuccessPayload;
+import org.eclipse.sirius.web.domain.boundedcontexts.representationdata.RepresentationIconURL;
+import org.eclipse.sirius.web.domain.boundedcontexts.representationdata.RepresentationMetadata;
+import org.eclipse.sirius.web.domain.boundedcontexts.representationdata.services.api.IRepresentationMetadataSearchService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import reactor.core.publisher.Sinks;
+
+/**
+ * Handler used to duplicate a representation.
+ *
+ * @author frouene
+ */
+@Service
+public class DuplicateRepresentationEventHandler implements IEditingContextEventHandler {
+
+    private static final String NEW_REPRESENTATION_METADATA = "newRepresentationMetadata";
+    private static final String COPY_SUFFIX = "_copy";
+
+    private final Logger logger = LoggerFactory.getLogger(DuplicateRepresentationEventHandler.class);
+
+    private final ICollaborativeMessageService messageService;
+
+    private final IRepresentationMetadataSearchService representationMetadataSearchService;
+
+    private final IRepresentationSearchService representationSearchService;
+
+    private final IRepresentationMetadataPersistenceService representationMetadataPersistenceService;
+
+    private final IRepresentationPersistenceService representationPersistenceService;
```

Comment:
Why are you using APIs dedicated to the need of Sirius Web instead of relying on `IRepresentation{Metadata|Content}{Search|Creation}Service`?